stablehlo/tests/CheckOps.cpp

```diff
@@ -97,34 +109,45 @@ static llvm::ErrorOr<std::string> getSerializedTensorPath(StringRef probeId,
   if (!metadataFile.is_open()) return llvm::errc::io_error;
 
   std::string line;
+  std::vector<std::string> fields(3);
 
   for (uint32_t match = 0; metadataFile >> line && match <= iteration;
        ++match) {
     auto pos = line.find(probeId);
 
-    if (pos != std::string::npos && match == iteration)
-      return line.substr(pos + probeId.size() + 1);
+    if (pos != std::string::npos && match == iteration) {
+      // Parse the record in the form of: probe_id,mlir_type,serialized_path
+      splitString(line, ',', fields.begin());
```

Comment:
We don't need a generic string splitting method. Let's rename this to:

```
if (pos != std::string::npos && match == iteration)
    return extractMetadata(line);
```

And can use `llvm::StringRef` and its built in `split` method to implement this.